CMakeLists.txt

```diff
@@ -8,24 +8,63 @@ endif ()
 project(marian CXX C)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(BUILD_ARCH native CACHE STRING "Compile for this CPU architecture.")
 
 # Custom CMake options
 option(COMPILE_CPU "Compile CPU version" ON)
 option(COMPILE_CUDA "Compile GPU version" ON)
 option(COMPILE_EXAMPLES "Compile examples" OFF)
 option(COMPILE_SERVER "Compile marian-server" OFF)
 option(COMPILE_TESTS "Compile tests" OFF)
+option(COMPILE_WASM "Compile marian as WASM target" OFF)
+option(COMPILE_DECODER_ONLY "Compile marian-decoder only" OFF)
+option(COMPILE_WITH_PTHREADS "Compile with pthreads support" ON)
 option(USE_CCACHE "Use ccache compiler cache (https://ccache.dev)" OFF)
 option(USE_CUDNN "Use CUDNN library" OFF)
 option(USE_DOXYGEN "Build documentation with Doxygen" ON)
 option(USE_FBGEMM "Use FBGEMM" OFF)
 option(USE_MKL "Compile with MKL support" ON)
+option(USE_WASM_COMPATIBLE_BLAS "Compile with a WASM compatible blas for decoder only builds" OFF)
```

Comment:
I think you should expose only `COMPILE_WASM` or something similar outside as an option in this PR, and set/unset the remaining features under `if(COMPILE_WASM)` and else given the supplied GitHub build file only tests it for WASM (and not the others).

Reply:
Done. All cmake options that are wasm specific are encapsulated under one cmake option (`COMPILE_WASM` in our case).